### DISCUSSION TITLE
Using enumerator value as event argument with Assert.Raises

### DISCUSSION THREAD

**Discussion by RealYHD** [xunit/xunit#2219] [Question] [answered]
Hi all, I'm trying to use XUnit to test if an event is fired where the argument for the event is an enumerator. Due to the generic type constraint of needing the event argument to inherit `EventArgs`, I can't use `Assert.Raises<ExampleEnum>(...)` as it says "there is no boxing conversion from `ExampleEnum` to `EventArgs`" which (I believe) makes sense as I don't believe enums inherit from EventArgs, which can't be done?

The delegate `EventHandler<TEventArgs>(...)` does not have such constraint.  My question is therefore in two parts:

1. Is this a problem with how I'm doing things in terms of convention or common practice? If so, what should I be changing?
2. Is this something that can potentially be changed with XUnit? Since I don't understand why such a constraint exists if the `EventHandler` delegate itself does not have such a constraint. I'm not opening an issue yet since there may be grounds for why things are done the way they are.

Thanks in advance!

**Comment by bradwilson** [comment, marked as answer]
This was a constraint up through .NET Framework 4.0, which appears to have been removed from 4.5. So the assertion constraint can be removed as well now. I've opened an issue: https://github.com/xunit/xunit/issues/2221

**Reply by RealYHD** [reply]
I see, thank you!
